src/main/scala/org/zalando/spearheads/innkeeper/api/JsonDtos.scala

```diff
@@ -2,59 +2,82 @@ package org.zalando.spearheads.innkeeper.api
 
 import java.time.LocalDateTime
 
-import org.zalando.spearheads.innkeeper.api.Endpoint.{ EndpointType, Https, Protocol, ReverseProxy }
-import org.zalando.spearheads.innkeeper.api.PathMatcher.MatcherType
-
 import scala.collection.immutable.Seq
 
 /**
  * @author dpersa
  */
-case class Route(id: Long, route: NewRoute, createdAt: LocalDateTime,
-                 deletedAt: Option[LocalDateTime] = None)
+sealed trait RouteName {
+  def name: String
+}
 
-case class Endpoint(hostname: String, path: Option[String] = None,
-                    port: Option[Int] = Some(443),
-                    protocol: Option[Protocol] = Some(Https),
-                    endpointType: Option[EndpointType] = Some(ReverseProxy))
+case class ValidRouteName private[api] (name: String) extends RouteName
 
-object Endpoint {
+case class InvalidRouteName private[api] (name: String) extends RouteName
```

Comment:
I want to be sure that only valid names get into the DB... and I didn't want to use the java style asserts :)